### PR TITLE
[test] Cut ~1m off scala tests by decreasing spark partitions in tests

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: hail/src/test/scala/is/hail/TestUtils.scala [hunk 1]
```diff
@@ -391,6 +391,11 @@ object TestUtils {
     }
   }
 
+  def assertAllEvalTo(xs: (IR, Any)*)(implicit execStrats: Set[ExecStrategy]): Unit = {
+    assertEvalsTo(MakeTuple.ordered(xs.map(_._1)), Row.fromSeq(xs.map(_._2)))
```
Comment (patrick-schultz) [inline hail/src/test/scala/is/hail/TestUtils.scala]: Good idea!

File: hail/src/main/scala/is/hail/check/Gen.scala [hunk 2]
```diff
@@ -448,7 +448,7 @@ object Gen {
   }
 
   def interestingPosInt: Gen[Int] = oneOfGen(
-      oneOf(1, 2, Int.MaxValue - 1, Int.MaxValue),
+      oneOf(4, Int.MaxValue - 1, Int.MaxValue),
```
Comment (patrick-schultz) [inline hail/src/main/scala/is/hail/check/Gen.scala]: 4 seems like a pretty "typical" value already covered by `choose(1, 100)`. Isn't 1 a more interesting int we want to be sure to pick?

File: hail/src/test/scala/is/hail/linalg/BlockMatrixSuite.scala [hunk 3]
```diff
@@ -444,6 +441,8 @@ class BlockMatrixSuite extends HailSuite {
     forAll(twoMultipliableBlockMatrices(nonExtremeDouble)) { case (l: BlockMatrix, r: BlockMatrix) =>
       l.cache()
       r.cache()
+      println(l.blocks.getNumPartitions)
+      println(r.blocks.getNumPartitions)
```
Comment (patrick-schultz) [inline hail/src/test/scala/is/hail/linalg/BlockMatrixSuite.scala]: delete?

File: hail/src/main/scala/is/hail/check/Gen.scala (hunk 2 above)
Comment (tpoterba) [reply hail/src/main/scala/is/hail/check/Gen.scala]: 1 is interesting, I suppose, but it leads to block matrices with tens of thousands of 1-byte partitions, which is slow :(

Comment (tpoterba) [reply hail/src/main/scala/is/hail/check/Gen.scala]: oh, I changed this generally. I just need to change the BM gen.

Comment (patrick-schultz) [reply hail/src/main/scala/is/hail/check/Gen.scala]: Ahh, that makes more sense!

**Pull request merged**